prov/efa/test/efa_unit_test_utils.h

```diff
@@ -0,0 +1,113 @@
+#include <cmocka.h>
+#include <infiniband/verbs.h>
+#include "efa.h"
+
+/**
+ * @brief Initialize a efa_cq struct
+ * @param[in] efa_cq A pointer to `struct efa_cq`.
+ * @returns Error code. 0 if successful, non-zero if failed.
+ */
+static int efa_cq_initialize(struct efa_cq *efa_cq)
```

Comment:
Can we use `fi_cq_open` to create a real cq?

Reply:
Same as above.

Reply:
all the other tests require devices any way, and there is no way to just run this test, so I say put it in for simplicity.

Reply:
I have a hard opinion about unit test, especially around mocking.

If possible we should make these test executable regardless of the platform, and each of our test unit should adhere to its respective test boundary. If we are testing function `void ABC(struct A, struct B, struct C)`, we should produce mocked `A, B, C` for the test, rather than introducing another dependency that is not out of the test boundary.

For example, when we test `ssize_t efa_cq_readfrom(struct fid_cq *cq_fid, void *buf, size_t count, fi_addr_t *src_addr)`, we should not introduce an external API, i.e. `fi_cq_open` which is outside our test boundary. If `fi_cq_open` is broken, that will cause `efa_cq_readfrom` tests to fail - which is against the rationale of unit testing.